docs/proposal-rose-suite-run.md

```diff
@@ -39,66 +39,237 @@ The remaining points can happen any time after 1) is complete although note that
+  * If the workflow is already running the alternative is to treat this as an
+    alias to `cylc unpause`.
+* Continue running the workflow if it was stopped before completion
+  (replaces `cylc restart`).
+* If, for some reason, you want to re-run a workflow from the beginning you will
+  be able to use `cylc clean` to clean out the log directory (and whatever other
+  directories you want removing) after which `cylc play` can be used.
+  Note that this means that `cylc play` must be able to use the command
+  line options from the previous run as stored in the private database.
+  There are 2 options:
+  * Require the use of a `--re-run` option to enable this
+    (otherwise `cylc play` will fail if it finds an existing private database).
+  * Do this automatically if no existing `log` directory is found.
+* Note that `cylc play` will always load the latest workflow definition found in
+  the run directory and will respect any command line options (which may alter
+  the workflow definition). Therefore, continuing a workflow using `cylc play`
+  effectively implies a reload.
+  * The processed `cylc.flow` files will be kept so there will be a record of
+    any changes
+    (see also [cylc-flow #3763](https://github.com/cylc/cylc-flow/issues/3763)).
+* Support Rose optional configuration via the environment variable
+  `ROSE_SUITE_OPT_CONF_KEYS` or the option `--rose-opt-conf-key=KEY`.
+
+`cylc reinstall` - new command to re-install workflows.
+* Very similar to the `cylc install` command except that this applies to a
+  previously installed workflow (i.e. you run this command in the directory of
+  the installed workflow, not in the source directory).
+* The rsync will use `--delete` to ensure that any previously installed files
+  which have been removed from the source directory also get removed from the
+  installed workflow.
+  * Note that this will have the effect of removing any files installed via
+    `rose-suite.conf`. They will get recreated by a subsequent reload or play but
+    could affect a running workflow.
+* Any options specified as part of the original install will be retained unless
+  overridden on the command line.
+* Consider a `--dry-run` option to report what would be changed?
+  * Does this replace `rose suite-cmp-vc`?
+
+`cylc reload` - no change from existing command.
+
+`cylc validate` - as per existing command with the following changes:
+* If validating a workflow installed with `cylc install` then use any relevant
+  command line options specified as part of the install unless overridden on the
+  command line.
+* Will work fine on source directories which include `rose-suite.conf` files.
+* No `--strict` mode
+  (naked dummy tasks no longer allowed - all tasks must be declared).
+  * This one is controversial and requires further discussion!
+* Cyclic graph validation can be quite slow and is currently included as part of
+  `--strict`. Add a `--no-cyclic-graph-validation` option to disable this?
```

Comment:
Or should cyclic graph detection be opt-in?

Reply:
Happy with that, we could also turn it on automatically if the number of tasks is less than, say 100, else print an info line with the `--cyclic-graph-validation` CLI opt to raise awareness.

Reply:
Issue created: cylc/cylc-flow#3869